pysidt/sidt.py

```diff
@@ -932,6 +946,529 @@ def regularize(self, data=None, check_data=True):
+        `iter_item_cap`: maximum number of structures the extension generation algorithm can send for expansion. Default is 100.
+        `fract_nodes_expand_per_iter`: fraction of nodes to split at each iteration. If 0, only 1 node will be split at each iteration.
+        `r`: atom types to generate extensions. If None, all atom types will be used.
+        `r_bonds`: bond types to generate extensions. If None, [1, 2, 3, 1.5, 4] will be used.
+        `r_un`: unpaired electrons to generate extensions. If None, [0, 1, 2, 3] will be used.
+        `r_site`: surface sites to generate extensions. If None, [] will be used.
+        `r_morph`: surface morphology to generate extensions. If None, [] will be used.
+        `fract_threshold_to_predict_true`: fraction of relevant structures that favor true classification at which True will be predicted, this helps the algorithm avoid either false negatives or false positives when one is significantly preferable over the other  
+    """
+    def __init__(
+        self,
+        decomposition,
+        root_group=None,
+        nodes=None,
+        initial_root_splits=None,
+        n_strucs_min=1,
+        iter_max=2,
+        iter_item_cap=100,
+        fract_nodes_expand_per_iter=0,
+        r=None,
+        r_bonds=None,
+        r_un=None,
+        r_site=None,
+        r_morph=None,
+        fract_threshold_to_predict_true=0.5,
+    ):
+        if nodes is None:
+            nodes = dict()
+        if r_bonds is None:
+            r_bonds = [1, 2, 3, 1.5, 4]
+        if r_un is None:
+            r_un = [0, 1, 2, 3]
+        if r_site is None:
+            r_site = []
+        if r_morph is None:
+            r_morph = []
+
+        super().__init__(
+            decomposition=decomposition,
+            root_group=root_group,
+            nodes=nodes,
+            initial_root_splits=initial_root_splits,
+            n_strucs_min=n_strucs_min,
+            iter_max=iter_max,
+            iter_item_cap=iter_item_cap,
+            fract_nodes_expand_per_iter=fract_nodes_expand_per_iter,
+            r=r,
+            r_bonds=r_bonds,
+            r_un=r_un,
+            r_site=r_site,
+            r_morph=r_morph,
+            )
+
+        self.fract_threshold_to_predict_true = fract_threshold_to_predict_true
+        self.max_accuracy = 0.0
+
+    def select_nodes(self):
+        """
+        Picks the node with the most decompositions who a change in the decomposition's classification might improve overall classification
+        adding multiple nodes would require the datum map variables to need updated before analyzing what data to split the node over as
+        those maps may change when a node is added
+        """
+        self.datum_truth_map = {datum:[getattr(n,"rule") for n in self.mol_node_maps[datum]["nodes"]] for datum in self.datums}
```

Comment:
Please run a formatter to catch all of these types of issues
```suggestion
        self.datum_truth_map = {datum: [getattr(n, "rule") for n in self.mol_node_maps[datum]["nodes"]] for datum in self.datums}
```